diff --git a/Makefile b/Makefile
index da17c1fe..2662e51c 100644
--- a/Makefile
+++ b/Makefile
@@ -178,7 +178,7 @@ openocd.checkout:
 	fi
 
 openocd.build: openocd.checkout
-	cd utils/openocd && ./bootstrap && ./configure --enable-jtag_dpi --prefix=$(INSTALL_DIR)/openocd && $(MAKE) && $(MAKE) install
+	cd utils/openocd && ./bootstrap && ./configure --disable-werror --enable-jtag_dpi --prefix=$(INSTALL_DIR)/openocd && $(MAKE) && $(MAKE) install
 
 openocd.clean:
 	rm -rf $(INSTALL_DIR)/openocd tools/openocd
diff --git a/gap8/gap8.mk b/gap8/gap8.mk
index f743412e..62f2ec5d 100644
--- a/gap8/gap8.mk
+++ b/gap8/gap8.mk
@@ -173,7 +173,7 @@ openocd.checkout:
 
 openocd.build: openocd.checkout
 	cd utils/gap8-openocd && ./bootstrap
-	cd utils/gap8-openocd && ./configure --prefix=$(INSTALL_DIR)/gap8-openocd --program-prefix=gap8-
+	cd utils/gap8-openocd && ./configure --disable-werror --prefix=$(INSTALL_DIR)/gap8-openocd --program-prefix=gap8-
 	cd utils/gap8-openocd && make -j install
 	mkdir -p $(INSTALL_DIR)/gap8-openocd/share/openocd/scripts/tcl
 	cd utils/openocd_tools && cp -r tcl/* $(INSTALL_DIR)/gap8-openocd/share/openocd/scripts/tcl
diff --git a/gvsoc/gvsoc/vp_models.mk b/gvsoc/gvsoc/vp_models.mk
index e37a542b..3b287615 100644
--- a/gvsoc/gvsoc/vp_models.mk
+++ b/gvsoc/gvsoc/vp_models.mk
@@ -30,9 +30,6 @@ VP_COMP_STD_LDFLAGS=-lpulpvp
 VP_COMP_DBG_LDFLAGS=-lpulpvp-debug
 VP_COMP_SV_LDFLAGS=-lpulpvp-sv
 
-VP_COMP_CFLAGS += -Werror -Wfatal-errors
-VP_COMP_LDFLAGS += -Werror -Wfatal-errors
-
 ifdef VP_USE_SYSTEMV
 VP_COMP_CFLAGS += -D__VP_USE_SYSTEMV
 endif
diff --git a/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/cluster/delegate/hyperbus/hyperbus_cl_internal.h b/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/cluster/delegate/hyperbus/hyperbus_cl_internal.h
index c6a517b8..a61d33cf 100644
--- a/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/cluster/delegate/hyperbus/hyperbus_cl_internal.h
+++ b/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/cluster/delegate/hyperbus/hyperbus_cl_internal.h
@@ -143,7 +143,7 @@ static inline void pi_cl_hyper_copy(struct pi_device *device, uint32_t hyper_add
                                     void *addr, uint32_t size,
                                     int ext2loc, pi_cl_hyper_req_t *req)
 {
-    __pi_cl_hyper_copy(device, hyper_addr, addr, size, !ext2loc, req);
+    __pi_cl_hyper_copy(device, hyper_addr, addr, size, (udma_channel_e)!ext2loc, req);
     hal_compiler_barrier();
 }
 
@@ -152,7 +152,7 @@ static inline void pi_cl_hyper_copy_2d(struct pi_device *device, uint32_t hyper_
                                        uint32_t stride, uint32_t length,
                                        int ext2loc, pi_cl_hyper_req_t *req)
 {
-    __pi_cl_hyper_copy_2d(device, hyper_addr, addr, size, stride, length, !ext2loc, req);
+    __pi_cl_hyper_copy_2d(device, hyper_addr, addr, size, stride, length, (udma_channel_e)!ext2loc, req);
 }
 
 static inline uint32_t pi_cl_hyperram_alloc_wait(pi_cl_hyperram_alloc_req_t *req)
diff --git a/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/drivers/fc_event/fc_event.h b/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/drivers/fc_event/fc_event.h
index 5918824d..46e900a7 100644
--- a/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/drivers/fc_event/fc_event.h
+++ b/rtos/freeRTOS/vendors/gwt/gap8/pmsis/include/pmsis/implem/drivers/fc_event/fc_event.h
@@ -33,6 +33,10 @@
 
 #include "pmsis.h"
 
+#if defined(__cplusplus)
+extern "C" {
+#endif /* __cplusplus */
+
 /*!
  * @addtogroup FC_EventHandler
  * @{
@@ -46,7 +50,7 @@
  * APIs
  ******************************************************************************/
 
-#define NB_SOC_EVENTS    ( SOC_EU_NB_FC_EVENTS )
+#define NB_SOC_EVENTS (SOC_EU_NB_FC_EVENTS)
 
 typedef void (*pi_fc_event_handler_t)(void* arg);
 
